### PR TITLE
CHEF-12173: Update regular expression to fix warnings detected by CodeQL

### PR DESCRIPTION
**Pull request by ahasunos**
## Description

This PR updates the regex used in our resources to fix the warnings detected by CodeQL.
 
## Related Issue

CHEF-12173: Polynomial regular expression used on uncontrolled data Chef InSpec

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New content (non-breaking change)
- [ ] Breaking change (a content change which would break existing functionality or processes)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: lib/inspec/utils/parser.rb [hunk 1]
```diff
@@ -72,15 +72,27 @@ def parse_mount_options(mount_line, compatibility = false)
         if includes_whitespaces?(mount_line)
           # Device-/Sharenames and Mountpoints including whitespaces require special treatment:
           # We use the keyword ' type ' to split up and rebuild the desired array of fields
+          # Split the mount line by the keyword ' type '
           type_split = mount_line.split(" type ")
           fs_path = type_split[0]
           other_opts = type_split[1]
```
Comment (sathish-progress) [inline lib/inspec/utils/parser.rb]: `fs_path, other_opts = mount_line.split(" type ", 2)`

File: lib/inspec/utils/parser.rb [hunk 2]
```diff
@@ -72,15 +72,27 @@ def parse_mount_options(mount_line, compatibility = false)
...
-          fs, path = fs_path.match(%r{^(.+?)\son\s(/.+?)$}).captures
+
+          # Manually split fs_path into the filesystem and path parts
+          parts = fs_path.split(" on ")
+          fs = parts[0]
+          path = parts[1]
```
Comment (sathish-progress) [inline lib/inspec/utils/parser.rb]: ` fs, path = fs_path.split(" on ", 2)`

File: lib/inspec/utils/parser.rb [hunk 3]
```diff
@@ -109,8 +121,16 @@ def parse_mount_options(mount_line, compatibility = false)
 
       # Device-/Sharename or Mountpoint includes whitespaces?
       def includes_whitespaces?(mount_line)
-        ws = mount_line.match(/^(.+)\son\s(.+)\stype\s.*$/)
-        ws.captures[0].include?(" ") || ws.captures[1].include?(" ")
+        # Split the mount_line by " on "
+        parts = mount_line.split(" on ")
+
+        # Check if the first part (device/share) or the second part (path) contains " type "
```
Comment (sathish-progress) [inline lib/inspec/utils/parser.rb]: where is this comment being satisfied?

File: lib/inspec/utils/parser.rb [hunk 4]
```diff
@@ -109,8 +121,16 @@ def parse_mount_options(mount_line, compatibility = false)
...
+        return false if parts.length != 2
+
+        fs_part, path_part = parts
+
+        # Check if either part contains spaces
+        fs_part.include?(" ") || path_part.include?(" ")
```
Comment (sathish-progress) [inline lib/inspec/utils/parser.rb]:
```suggestion
parts.any? { |part| part.include?(" ") }
```

File: lib/inspec/utils/parser.rb (hunk 3 above)
Comment (ahasunos) [reply lib/inspec/utils/parser.rb]: please ignore this. Forgot to remove this during the cleanup.

File: lib/inspec/utils/parser.rb [hunk 5]
```diff
@@ -109,8 +121,16 @@ def parse_mount_options(mount_line, compatibility = false)
 
       # Device-/Sharename or Mountpoint includes whitespaces?
       def includes_whitespaces?(mount_line)
-        ws = mount_line.match(/^(.+)\son\s(.+)\stype\s.*$/)
-        ws.captures[0].include?(" ") || ws.captures[1].include?(" ")
+        # Split the mount_line by " on "
+        parts = mount_line.split(" on ")
+
+        # Check if the first part (device/share) or the second part (path) contains " type "
+        return false if parts.length != 2
```
Comment (sathish-progress) [inline lib/inspec/utils/parser.rb]: I guess you are assuming that if there is only 1 string then white space would not be present.. but the following has trailing white spaces.
```
'aa  on '.split(" on ")
=> ["aa "]
```

And why the parts should not be more than 2 ?

Comment (ahasunos) [reply lib/inspec/utils/parser.rb]: examples of mount_line
`"//fileserver.corp.internal/Research & Development on /mnt/Research & Development type cifs (rw,vers=1.0)\n"`

`"/dev/xvda1 on / type ext4 (rw,discard)\n"`

Comment (sathish-progress) [reply lib/inspec/utils/parser.rb]: exactly, you could have the mount line with custom names, 
for example, see the below values and their behaviour
```
parts = "//fileserver.corp.internal/Research & Development on server   on /mnt/Research & Development type cifs (rw,vers=1.0)\n".split(" on ")
=> ["//fileserver.corp.internal/Research & Development", "server  ", "/mnt/Research & Development type cifs (rw,vers=1.0)\n"]
```

**Review by sathish-progress — APPROVED** [review]

**Pull request merged**